### PR TITLE
Refactor apis/run to standalone go module

### LINKED ISSUE #3174: Refactor apis/run to its own go module
**Describe the feature request**

As part of tanzu-framework refactoring we are breaking up repo into multiple go modules. It was decided to break up apis/run into its own go module

**Affected product area (please put an X in all that apply)**

- (x) APIs
- ( ) Addons
- ( ) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Additional context**

### PR DESCRIPTION
**Pull request by vijaykatam**
Signed-off-by: Vijay Katam <vkatam@vmware.com>

### What this PR does / why we need it

As part of tanzu-framework refactoring we are breaking up repo into multiple go modules. It was decided to break up apis/run into its own go module

### Which issue(s) this PR fixes

Fixes #3174 

### Describe testing done for PR
Unit tests passing in CI

### Release note

```release-note
None
```

### CODE REVIEW

File: apis/run/Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2022 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+include ../../common.mk
+# set root for manifests
+RELATIVE_ROOT=.
+
+.PHONY: manifests
+manifests: ## Generate manifests e.g. CRD, RBAC etc.
+	$(MAKE) generate-manifests CONTROLLER_GEN_SRC=./...
```
Comment (codegold79) [inline apis/run/Makefile:10]: nit: probably should have newline

**Review by codegold79 — APPROVED** [review]
As long as everything still works, LGTM. I can't see anything wrong with the changes.

File: pkg/v2/tkr/util/sets/compatibility.go [hunk 2]
```diff
@@ -1,14 +1,12 @@
 // Copyright 2022 VMware, Inc. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-package version
+package sets
```
Comment (imikushin) [inline pkg/v2/tkr/util/sets/compatibility.go:4]: I'd like to move this package (`sets`) to `apis/run/util` and keep this `Compatibility` interface in the `version` package.

Comment (vijaykatam) [reply pkg/v2/tkr/util/sets/compatibility.go:4]: Sure, let me look at making this change.

File: apis/run/Makefile [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Copyright 2022 VMware, Inc. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+include ../../common.mk
+# set root for manifests
+RELATIVE_ROOT=.
```
Comment (maralavi) [inline apis/run/Makefile:6]: Just wondering why we need to define ```RELATIVE_ROOT```?

Comment (vijaykatam) [reply apis/run/Makefile:6]: make target `generate-manifests` in common.mk is using it for output folder for crds

**Review by imikushin — APPROVED** [review]
Looks good. Thanks @vijaykatam!

**Pull request merged**